pkg/webhook/listener/webhook_listner.go

```diff
@@ -164,9 +178,112 @@ func CreateWebhookListener(config, remoteConfig *rest.Config, scheme *runtime.Sc
 		}
 	}
 
+	if createService {
+		namespace, err := k8sutil.GetOperatorNamespace()
+
+		if err != nil {
+			return nil, err
+		}
+
+		// Create the webhook listener service only when the subscription controller runs in hub mode.
+		err = createWebhookListnerService(l.LocalClient, namespace)
+
+		if err != nil {
+			klog.Error("Failed to create a service for Git webhook listener. error: ", err)
+			return nil, err
+		}
+	}
+
 	return l, err
 }
 
+func createWebhookListnerService(client client.Client, namespace string) error {
+	var theServiceKey = types.NamespacedName{
+		Name:      serviceName,
+		Namespace: namespace,
+	}
+
+	service := &corev1.Service{}
+
+	if err := client.Get(context.TODO(), theServiceKey, service); err != nil {
```

Comment:
not sure if we need a flow to detect and update the existing service, to make sure the leftover one is also in the desired state.

Reply:
@ianzhang366 I am not sure what you mean by the leftover one. When ACM is uninstalled, the hub-subscription pod gets deleted which deletes this service due to owner ref.

Reply:
@ianzhang366 I am not sure what you mean by the leftover one. When ACM is uninstalled, the hub-subscription pod gets deleted which deletes this service due to owner ref.

Reply:
I was thinking the case, that if we patch something to the pod and the service is different then the current logic won't pick the new changes, not sure this is a valid flow you want to look into tho.

Reply:
You are right that there could be some change in the service configuration but I do not foresee any change in the near future. If there is change, it should be dealt with at that time. I will go ahead with only necessary code for now. Thanks.